src/web/mjs/connectors/WestManga.mjs

```diff
@@ -1,13 +1,123 @@
-import WordPressMangastream from './templates/WordPressMangastream.mjs';
-
-export default class WestManga extends WordPressMangastream {
+import Connector from "../engine/Connector.mjs";
+import Manga from "../engine/Manga.mjs";
 
+export default class WestManga extends Connector {
     constructor() {
         super();
-        super.id = 'westmanga';
-        super.label = 'WestManga';
-        this.tags = [ 'manga', 'indonesian' ];
-        this.url = 'https://westmanga.me';
-        this.path = '/manga/list-mode/';
+        super.id = "westmanga";
+        super.label = "WestManga";
+        this.tags = ["manga", "manhua", "manhwa", "indonesian"];
+        this.url = "https://westmanga.me";
+        this.queryMangaTitle = "h1";
+
+        // API configuration
+        this.api = {
+            url: "https://data.westmanga.me/api/",
+            nonce: "wm-api-request",
+            accessKey: "WM_WEB_FRONT_END",
+            secretKey: "xxxoidj",
+        };
+    }
+
+    cleanTitle(title) {
+        return title.replace(/bahasa indonesia/i, "").trim();
+    }
+
+    async _getMangas() {
+        const mangaList = [];
+        for (let page = 1; ; page++) {
+            const mangas = await this.getMangasFromPage(page);
+            if (mangas.length === 0) break;
+            mangaList.push(...mangas);
+        }
+        return mangaList;
+    }
+
+    async getMangasFromPage(page) {
+        const { data } = await this.fetchAPI(`./contents?page=${page}`);
+        return data
+            ? data.map(({ slug, title }) => ({
```

Comment:
I am not sure that Hakuneko stable 6.1.7 is ok with the ? : syntax . Perhaps i am just too careful.